pkg/codegen/pcl/program.go

```diff
@@ -169,6 +171,47 @@ func (p *Program) Source() map[string]string {
 	return source
 }
 
+// writeSourceFiles writes the source files of the program, including those files of used components into the
+// provided file system starting from a root directory.
+func (p *Program) writeSourceFiles(directory string, fs afero.Fs, seenPaths map[string]bool) error {
```

Comment:
Note that even if we go with just the Fs paramter and using BasePathFs for `WriteSourceFiles` this internal method would I think still need the directory parameter because it may need "../" style paths.

Reply:
I need `directory` not only for the root working directory but also when traversing directories of components, so providing a root dir seems redundant for afero